snakefmt/snakefmt.py

```diff
@@ -111,10 +117,9 @@ def get_snakefiles_in_dir(
 @click.option(
     "-l",
     "--line-length",
-    default=DEFAULT_LINE_LENGTH,
-    show_default=True,
+    default=None,
     type=int,
-    help="Lines longer than INT will be wrapped.",
+    help=f"Lines longer than INT will be wrapped. [default: {DEFAULT_LINE_LENGTH}]",
```

Comment:
Why has this been changed? This creates more work as we now have to create the CLI default string ourselves? If there is a default then we should declare it here?

Reply:
The issue I try to solve is the following: `snakefmt` toml-defined line_length overrides `black` toml-define line length. If we always pass `DEFAULT_LINE_LENGTH` to `Formatter`, even when it has not been specified (at CLI or in toml), then `DEFAULT_LINE_LENGTH` will be used instead of any value used in `black` toml. Eg line_length is 50 is pyproject.toml under [tool.black], that currently only is registered if default is None. But I still want to explain that when no config is used, we use that default. I agree it's not the most elegant, but could not find a better, easily functional way.

Reply:
Hmm, I see. I'm starting to think we may need a FileMode-like class that extends black`s so we don't need to think about line length in two places.

Reply:
Feel free to propose a change, I'm personally happy with it like this. I don't think we should extend `FileMode` in `snakefmt.py`, black configuration is processed/used in `formatter.py`

Reply:
I wasn't suggesting we do it here - it's outside of the scope of this PR. More a discussion point.